pylint/checkers/strings.py

```diff
@@ -395,10 +398,54 @@ def visit_binop(self, node: nodes.BinOp) -> None:
                             args=(arg_type.pytype(), format_type),
                         )
 
-    @only_required_for_messages("f-string-without-interpolation")
+    @only_required_for_messages(
+        "f-string-without-interpolation", "bad-format-character"
```

Comment:
All messages emitted by a `visit_*` or `leave_*` callback must be included in the `@only_required_for_messages` decorator. `bad-format-string` and `bad-string-format-type` are currently missing, which is the reason why there are failed tests in the CI pipeline.